src/components/FileManager.vue

```diff
@@ -0,0 +1,290 @@
+<template>
+    <drop-area
+        @newFiles="uploadFiles">
+      <div class="drop-area-title">
+        <slot name="dropAreaTitle">
+          <h3>
+            Upload new file
+          </h3>
+        </slot>
+      </div>
+    <div class="documents-container">
+      <div class="documents">
+        <div
+          v-for="upload in uploads"
+          v-bind:key="upload.name"
+          class="document"
+          :class="{'error': upload.retry >= maxUploadRetries}"
+          >
+          <div v-if='upload.retry >= maxUploadRetries'
+            class="document-delete"
+            @click="deleteUpload(upload)"
+            >
+            <DeleteIcon class="delete-icon" />
+          </div>
+          <div class="document-loading" v-if='upload.uploadPercentage < 100'>
+            <div
+              v-if='upload.uploadPercentage < 100'
+              :style="{'width':  Math.floor(upload.uploadPercentage) + '%'}"
+              class="document-loading-loading"
+              >
+              </div>
+            </div>
+            <div class="document-image">
+              <GenericFileImage class="generic-file-image" />
+            </div>
+              <div class="document-name">
+                  {{ upload.name }}
+              </div>
+          </div>
+
+          <div
+            v-for="document in orderedDocuments"
+            v-bind:key="document.name"
+            class="document"
+            >
+              <div
+                class="document-delete"
+                @click="deleteDocument(document)"
+                >
+                  <DeleteIcon class="delete-icon" />
+              </div>
+              <div class="document-image">
+                    <img v-if="document.thumbnail" :src="document.thumbnail">
+                    <GenericFileImage v-else class="generic-file-image" />
+              </div>
+              <div class="document-name">
+                {{ document.name }}
+              </div>
+            </div>
+
+      </div>
+    </div>
+    </drop-area>
+</template>
+
+<script>
+import _ from 'lodash'
+import API from '../api.js'
+import DropArea from './DropArea.vue'
+import DeleteIcon from './../assets/cancel.svg'
+import GenericFileImage from './../assets/file.svg'
+
+export default {
+  name: 'filemanager',
+  components: {
+    DropArea,
+    DeleteIcon,
+    GenericFileImage
+  },
+  props: ['apiUrl', 'maxUploadRetries'],
+  data () {
+    return {
+      uploads: [],
+      documents: []
+    }
+  },
+  computed: {
+    orderedDocuments () {
+      return _.orderBy(this.documents, 'name', 'asc')
+    }
+  },
+  created () {
+    if (this.maxUploadRetries === undefined || this.maxUploadRetries === null) {
+      this.maxUploadRetries = 3
+    }
+    API.setUrlAPI(this.apiUrl)
+    this.getDocuments()
+    let self = this
+    setInterval(function () {
```

Comment:
Use arrow function to avoid using self. Why do we need to refresh every 30s?

Reply:
To get documents synchronized with the server